server/src/service-layer/controllers/PaymentController.ts

```diff
@@ -176,4 +184,182 @@ export class PaymentController extends Controller {
+    ) {
+      this.setStatus(401)
+      return {
+        error:
+          "Invalid date, booking start date and end date must be in the range of today up to a year later. "
+      }
+    }
+    // Calculate number of days of booking, returns a 0 indexed number
+    function getNumberOfDays(startDate: Timestamp, endDate: Timestamp): number {
+      const MILLIS_IN_A_DAY = 1000 * 60 * 60 * 24
+      const numberOfDays =
+        (startDate.seconds * 1000 - endDate.seconds * 1000) / MILLIS_IN_A_DAY
+      return numberOfDays + 1
+    }
+    // Calculate number of days of booking
+    const numberOfDays = getNumberOfDays(startDate, endDate)
+    // Validate number of dates to avoid kiddies from forging bookings
+    if (numberOfDays > 10) {
+      this.setStatus(401)
+      return {
+        error: "Invalid date range, booking must be a maximum of 10 days. "
+      }
+    }
+    // Create new Stripe checkout session
+    const stripeService = new StripeService()
+    const userDataService = new UserDataService()
+
+    const userData = await userDataService.getUserData(uid)
+    const { newUser, stripeCustomerId } =
+      await stripeService.createCustomerIfNotExist(
+        request.user,
+        userData,
+        userDataService
+      )
+    // If not a new Stripe customer, we want to check for pre-existing bookings
+    if (!newUser) {
+      const activeSession = await stripeService.getActiveSessionForUser(
+        stripeCustomerId,
+        CheckoutTypeValues.BOOKING
+      )
+      if (activeSession) {
+        return {
+          stripeClientSecret: activeSession.client_secret,
+          message: "Existing booking checkout session found"
+        }
+      }
+    }
+
+    const recentActiveSessions = await stripeService.getRecentActiveSessions(
+      CheckoutTypeValues.BOOKING,
+      30
+    )
+
+    const bookingSlotService = new BookingSlotService()
+    const bookingDataService = new BookingDataService()
+
+    const bookingSlots =
+      await bookingSlotService.getBookingSlotsBetweenDateRange(
+        startDate,
+        endDate
+      )
+    // Create a date range list to validate against
+    const daysList = datesToDateRange(
+      firestoreTimestampToDate(startDate),
+      firestoreTimestampToDate(endDate)
+    )
+
+    if (bookingSlots.length !== daysList.length) {
+      this.setStatus(401)
+      return {
+        error: "No booking slot available for one or more dates."
+      }
+    }
+    // iterate through the active stripe sessions and check metadata, reduce the bookingSlots available
+    for (const session of recentActiveSessions) {
+      const bookingSlotMetadata = session.metadata.booking_slots
+      if (bookingSlotMetadata) {
+        for (const bookingSlotId of bookingSlotMetadata) {
+          const bookingSlot = bookingSlots.find(
+            (slot) => slot.id === bookingSlotId
+          )
+          if (bookingSlot) {
+            bookingSlot.max_bookings -= 1
+          }
+        }
+      }
+    }
```

Comment:
We don't need this because reducing the booking slots is handled in the backend webhook in #370